tests/openedx_tagging/core/tagging/test_system_defined.py

```diff
@@ -0,0 +1,236 @@
+
+from .test_models import TestTagTaxonomyMixin
+
+test_languages = [
+    ('en', 'English'),
+    ('az', 'Azerbaijani'),
+    ('id', 'Indonesian'),
+    ('qu', 'Quechua'),
+    ('zu', 'Zulu'),
+]
+
+
+class EmptyTestClass:
+    """
+    Empty class used for testing
+    """
+
+
+class EmptyTestModel(models.Model):
+    """
+    Model used for testing
+    """
+    mi_id = models.AutoField(primary_key=True)
+
+    class Meta:
+        managed = False
+        app_label = 'oel_tagging'
+
+
+class EmptyModelObjectTag(ModelObjectTag):
+    """
+    Model ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+
+class NotDjangoModelObjectTag(ModelObjectTag):
+    """
+    Model ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+    tag_class_model = EmptyTestClass
+
+
+class NotIdModelObjectTag(ModelObjectTag):
+    """
+    Model ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+    tag_class_model = EmptyTestModel
+
+
+class TestUserObjectTag(UserObjectTag):
+    """
+    User ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+
+@ddt.ddt
+class TestSystemDefinedObjectTags(TestTagTaxonomyMixin, TestCase):
+    """
+    Test for generic system defined object tags
+    """
+    def test_system_defined_is_valid(self):
+        # Valid
+        assert TestUserObjectTag()._check_system_taxonomy(taxonomy=self.user_system_taxonomy)
+
+        # Null taxonomy
+        assert not UserObjectTag()._check_system_taxonomy()
+
+        # Not system defined taxonomy
+        assert not UserObjectTag()._check_system_taxonomy(taxonomy=self.taxonomy)
+
+        # Not connected with the taxonomy
+        assert not UserObjectTag()._check_system_taxonomy(taxonomy=self.user_system_taxonomy)
+
+    @ddt.data(
+        (EmptyModelObjectTag, False),  # Without associated model
+        (NotDjangoModelObjectTag, False),  # Associated class is not a Django model
+        (NotIdModelObjectTag, False),  # Associated model has not 'id' field
+        (ModelObjectTag, False), # Testing parent class validations
+        (TestUserObjectTag, True),  #Valid
+    )
+    @ddt.unpack
+    def test_model_object_is_valid(self, object_tag, assert_value):
+        args = {
+            'taxonomy': self.user_system_taxonomy,
+            'object_id': 'id',
+            'object_type': 'object',
+            'value': 'value',
+        }
+        result = object_tag(**args).is_valid(check_object=False, check_tag=False, check_taxonomy=True)
+        self.assertEqual(result, assert_value)
```

Comment:
nit: can do this with a plain assert check:
```suggestion
        assert result == assert_value
```

Reply:
All test has been updated with this